### ISSUE TITLE
Long Pandas compile time (>7 minutes)

### ISSUE THREAD

**Issue by beoRich** [astral-sh/uv#8885]
We are trying uv as pip replacement and encounter an usual long compile time:
uv sync --reinstall -n --verbose

```
.....
DEBUG copying pandas\conftest.py -> build\lib.win-amd64-cpython-312\pandas
DEBUG copying pandas\testing.py -> build\lib.win-amd64-cpython-312\pandas
DEBUG copying pandas\_typing.py -> build\lib.win-amd64-cpython-312\pandas
DEBUG copying pandas\_version.py -> build\lib.win-amd64-cpython-312\pandas
DEBUG copying pandas\__init__.py -> build\lib.win-amd64-cpython-312\pandas
DEBUG creating build\lib.win-amd64-cpython-312\pandas\api
DEBUG copying pandas\api\__init__.py -> build\lib.win-amd64-cpython-312\pandas\api
DEBUG creating build\lib.win-amd64-cpython-312\pandas\arrays
DEBUG copying pandas\arrays\__init__.py -> build\lib.win-amd64-cpython-312\pandas\arrays
DEBUG creating build\lib.win-amd64-cpython-312\pandas\compat
DEBUG copying pandas\compat\chainmap.py -> build\lib.win-amd64-cpython-312\pandas\compat
DEBUG copying pandas\compat\pickle_compat.py -> build\lib.win-amd64-cpython-312\pandas\compat
DEBUG copying pandas\compat\pyarrow.py -> build\lib.win-amd64-cpython-312\pandas\compat
.....
```

and on via dockerimage:

```
# builder image
FROM ghcr.io/astral-sh/uv:python3.12-bookworm AS builder
COPY --from=ghcr.io/astral-sh/uv:latest /uv /uvx /bin/

WORKDIR /app

ENV UV_LINK_MODE=copy
ENV UV_COMPILE_BYTECODE=0
ENV UV_SYSTEM_PYTHON=1

RUN --mount=type=cache,target=/root/.cache/uv \
    --mount=type=bind,source=uv.lock,target=uv.lock \
    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
    uv sync --verbose --frozen --no-install-project --no-dev

ADD . /app
RUN --mount=type=cache,target=/root/.cache/uv \
    uv sync --frozen --no-dev

# final image
FROM python:3.12-slim-bookworm
COPY --from=builder --chown=app:app /app /app
WORKDIR /app

ENTRYPOINT ["python3"]
CMD ["-m", "application"]
```

Since we are using a CI/CD Pipeline with no cache this is too long to be useable.
Pipeline CD Step with uv, dockerbuildtime: 645.7s
Pipeline CD Step with pip, dockerbuildtime: around 90s

Is there any way to fix this?

**Comment by charliermarsh** [comment]
I don't think using pip vs. uv should have any impact on Pandas build times -- they both just call the relevant build backend, none of the build happens in pip or uv. I have two questions: (1) why is Pandas building from source in the first place? Are you on a really old version that doesn't ship wheels or something? And (2) are you sure that you're also building Pandas from source in the pip version of your pipeline?

**Comment by beoRich** [comment]
1) Don't think so, the version ist 1.5.3 which was released Jan 19, 2023. We have no idea why its building from source...it is added via transitive dependency via an own package with extra Index.
project -> own package -> own package (with pandas 1.5.3)
The packages itself are published via pip, setup.py and twine:
```
python setup.py sdist bdist_wheel
twine upload --repository codeartifact dist/*
```
2) Yes we saw the same logs there.
